src/ArborX_LinearBVH.hpp

```diff
@@ -147,38 +149,52 @@ BoundingVolumeHierarchy<DeviceType>::BoundingVolumeHierarchy(
     if ( size() == 1 )
     {
         Kokkos::View<size_t *, DeviceType> permutation_indices( "permute", 1 );
+        Kokkos::Profiling::pushRegion( "ArborX:BVH:initialize_leaf_nodes" );
         Details::TreeConstruction<DeviceType>::initializeLeafNodes(
             primitives, permutation_indices, getLeafNodes() );
+        Kokkos::Profiling::popRegion();
         return;
     }
 
     // determine the bounding box of the scene
+    Kokkos::Profiling::pushRegion( "ArborX:BVH:calculate_scene_bounding_box" );
     Details::TreeConstruction<DeviceType>::calculateBoundingBoxOfTheScene(
         primitives, getBoundingVolume( getRoot() ) );
+    Kokkos::Profiling::popRegion();
 
     // calculate morton code of all objects
     Kokkos::View<unsigned int *, DeviceType> morton_indices(
         Kokkos::ViewAllocateWithoutInitializing( "morton" ), size() );
+    Kokkos::Profiling::pushRegion( "ArborX:BVH:assign_morton_codes" );
```

Comment:
Did you mean to exclude the allocation of the array?

Reply:
I don't think it matters, but I'm OK moving it up.